cpp/src/gandiva/function_registry_string.cc

```diff
@@ -222,13 +222,13 @@ std::vector<NativeFunction> GetStringFunctionRegistry() {
                      NativeFunction::kCanReturnErrors),
 
       NativeFunction("parse_url", {}, DataTypeVector{utf8(), utf8()}, utf8(),
-                     kResultNullIfNull, "gdv_fn_parse_url_utf8_utf8",
+                     kResultNullInternal, "gdv_fn_parse_url_utf8_utf8",
```

Comment:
Thanks for your fix. This need to be set if we want use `out_valid` in function. I've forget to update this change, thanks for your fix.